protos/core.proto

```diff
@@ -250,7 +250,12 @@ message NotebookRuntimeOptions {
 
 // Data Preparation Related entries
 message DataPreparation {
-  // Data preparatiohs can have more than 1 output
+  // Used ONLY as an identifiers. All outputs should be listed in the targets
+  // section.
+  Target target = 8;
+  Target canonical_target = 9;
+
+  // Data preparations can have more than 1 output
```

Comment:
These should be put in the `configs.proto` file too, or they won't be user overrideable

Reply:
Given that the data preparation is self-containing, I don't think this is something that we should allow the user to override.